src/gc/gc.cpp

```diff
@@ -15357,6 +15377,11 @@ void gc_heap::gc1()
     assert (g_gc_card_bundle_table == card_bundle_table);
 #endif    
 
+    if (!g_gc_pStringClass)
+    {
+        g_gc_pStringClass = GCToEEInterface::GetStringMethodTable();
```

Comment:
This is the logically closest place before actual usage of g_gc_pStringClass.
But we may put it higher on the callstack ie closer to collection begining.